packages/@glimmer/runtime/lib/opcodes.ts

```diff
@@ -60,126 +61,154 @@ export class AppendOpcodes {
     evaluate: Syscall | MachineOpcode,
     kind = 'syscall'
   ): void {
-    this.evaluateOpcode[name as number] = {
+    this.#evaluateOpcode[name as number] = {
       syscall: kind !== 'machine',
       evaluate,
     } as Evaluate;
   }
 
-  debugBefore(vm: VM, opcode: RuntimeOp): DebugState {
-    let params: Maybe<Dict> = undefined;
-    let opName: string | undefined = undefined;
-
-    if (LOCAL_SHOULD_LOG) {
-      let pos = vm[INNER_VM].fetchRegister($pc) - opcode.size;
-
-      [opName, params] = debug(vm[CONSTANTS], opcode, opcode.isMachine)!;
-
-      // console.log(`${typePos(vm['pc'])}.`);
-      LOCAL_LOGGER.log(`${pos}. ${logOpcode(opName, params)}`);
-
-      let debugParams = [];
-      for (let prop in params) {
-        debugParams.push(prop, '=', params[prop]);
-      }
-
-      LOCAL_LOGGER.log(...debugParams);
+  static {
+    if (import.meta.env.DEV) {
```

Comment:
:raised_hands:

Reply:
CONFIRM. I've found a bunch of better patterns for stripping stuff that work better with TypeScript once you can assume rollup and terser optimizations.